server/src/test/java/io/crate/integrationtests/InsertIntoIntegrationTest.java

```diff
@@ -438,6 +438,9 @@ public void testInsertWithClusteredByNull() throws Exception {
             .hasPGError(INTERNAL_ERROR)
             .hasHTTPError(BAD_REQUEST, 4000)
             .hasMessageContaining("Clustered by value must not be NULL");
+
+        execute("insert into quotes (id, quote) select null, 'test'");
+        assertThat(response.rowCount()).isEqualTo(0L);
```

Comment:
Both insert-from-values and insert-from-subquery are covered by `testInsertWithClusteredByNull`.

I haven't added a check for "bad partitions are not created". 
This is possible on master if we do smth like
`insert into  (cl_by, parted_by) select null, 1` but here it's clear that `setNextRow` is called in grouper before creating an item.

Reply:
Please create a dedicated test case